### PR TITLE
Add docs for the `yaml` env var processor

### PR DESCRIPTION
**Pull request by mpdude**
For symfony/symfony#30910. Based on the file/json examples.

#FOSSHackathon

### CODE REVIEW

File: configuration/environment_variables.rst [hunk 1]
```diff
@@ -354,6 +354,43 @@ Symfony provides the following env var processors:
                 'trusted_hosts' => '%env(json:TRUSTED_HOSTS)%',
             ]);
 
+``env(yaml:FOO)``
```
Comment (OskarStark) [inline configuration/environment_variables.rst]: please add a `versionadded` directive

File: configuration/environment_variables.rst [hunk 2]
```diff
@@ -354,6 +354,43 @@ Symfony provides the following env var processors:
...
+    Decodes the content of ``FOO``, which is a YAML encoded string. It returns
+    either an array or ``null``. Mostly useful in combination with
+    ``file``:
+
+        .. code-block:: yaml
+
+            # config/services.yaml
+            parameters:
+                env(SECRETS_FILE): '/opt/application/.secrets.yaml'
+                database_password: '%env(key:database_password:yaml:file:SECRETS_FILE)%'
```
Comment (OskarStark) [inline configuration/environment_variables.rst]: could we change the example and remove the key:database_password processor here?

Comment (mpdude) [reply configuration/environment_variables.rst]: @OskarStark To make the example more concise? 

Would you ever use an array value as such?

File: configuration/environment_variables.rst (hunk 1 above)
Comment (mpdude) [reply configuration/environment_variables.rst]: done

File: configuration/environment_variables.rst (hunk 2 above)
Comment (OskarStark) [reply configuration/environment_variables.rst]: Hmm not sure, I haven’t any other example

**Comment by mpdude** [comment]
Closing as @nicolas-grekas and I agreed that symfony/symfony#30910 is not worthwhile.

**Pull request closed without merging**